File: AspNetMvcUnleashed/0672329980_source/Chapter08/VB/MvcApplication1/MvcApplication1/Models/Product.vb
Imports System.ComponentModel

Partial Public Class Product
    Implements IDataErrorInfo

    Private _errors As New Dictionary(Of String, String)()

    Private Sub OnNameChanging(ByVal value As String)
        If value.Trim() = String.Empty Then
            _errors.Add("Name", "Name is required.")
        End If
    End Sub

    Private Sub OnPriceChanging(ByVal value As Decimal)
        If value <= 0D Then
            _errors.Add("Price", "Price must be greater than 0.")
        End If
    End Sub

#Region "IDataErrorInfo Members"

    Public ReadOnly Property [Error]() As String Implements IDataErrorInfo.Error
        Get
            Return String.Empty
        End Get
    End Property

    Public ReadOnly Property Item(ByVal columnName As String) As String Implements IDataErrorInfo.Item
        Get
            If _errors.ContainsKey(columnName) Then
                Return _errors(columnName)
            End If
            Return String.Empty
        End Get
    End Property

#End Region


End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter08/CS/MvcApplication1/MvcApplication1/Models/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;

namespace MvcApplication1.Models
{
    public partial class Product : IDataErrorInfo
    {

        private Dictionary<string, string> _errors = new Dictionary<string, string>();

        partial void OnNameChanging(string value)
        {
            if (value.Trim() == String.Empty)
                _errors.Add("Name", "Name is required.");
        }


        partial void OnPriceChanging(decimal value)
        {
            if (value <= 0m)
                _errors.Add("Price", "Price must be greater than 0.");
        }


        #region IDataErrorInfo Members

        public string Error
        {
            get { return string.Empty; }
        }

        public string this[string columnName]
        {
            get
            {
                if (_errors.ContainsKey(columnName))
                    return _errors[columnName];
                return string.Empty;
            }
        }

        #endregion


    }
}
